ui/src/components/JsonDisplay/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { string, object } from 'prop-types';
+import { dump } from 'js-yaml';
+import { Switch, Typography, Grid } from '@material-ui/core';
+import Code from '../Code';
+import 'highlight.js/styles/atom-one-dark.css';
+
+const toYaml = obj => {
+  return dump(obj);
+};
+
+const toJson = obj => {
+  return JSON.stringify(obj, null, 2);
+};
+
+const YamlJsonSwitch = ({ handleChange, value }) => {
+  return (
+    <Typography component="div">
+      <Grid component="label" container alignItems="center" spacing={1}>
+        <Grid item>Yaml</Grid>
+        <Grid item>
+          <Switch
+            checked={value === 'yaml'}
+            onChange={handleChange}
+            name="yaml_json_switch"
+          />
+        </Grid>
+        <Grid item>Json</Grid>
+      </Grid>
+    </Typography>
+  );
+};
+
+const JsonDisplay = props => {
+  const [language, setLanguage] = useState(props.language);
+  const toText = language === 'yaml' ? toYaml : toJson;
+
+  return (
+    <React.Fragment>
+      <YamlJsonSwitch
+        value={language}
+        handleChange={() => {
+          setLanguage(language === 'yaml' ? 'json' : 'yaml');
+        }}
+      />
+      <Code {...props}>{toText(props.toDisplay)}</Code>
+    </React.Fragment>
+  );
+};
+
+JsonDisplay.propTypes = {
+  /**
+   * The content object to be serialized to yaml/json.
+   */
+  toDisplay: object.isRequired,
+  /**
+   * A highlight.js language identifier.
+   */
+  language: string.isRequired,
+  /** The CSS class name of the wrapper element */
+  className: string,
```

Comment:
I think it'd be cleaner to rename the variables/prop names so they are more descriptive, rather than naming them generically then adding comments :)

Reply:
I reused properties from the Code block so I won't have to change it in all places the JsonDisplay is used

Reply:
That reasoning doesn't make sense to me. You've already had to update files due to the component name changing, so there's no reason not to refactor the prop names. It's cleaner and a better practice to get into.

Reply:
So please either remove the redundant comments or rename props accordingly.

Reply:
yes you are right, I thought I just changed the import name will change them in that case

Reply:
I agree, if this new component is going to replace all the Code instances, then it is appropriate to rename some properties to better fit the new usage. I'd support removing unused properties as well, if any no longer make sense or are unused by current instances.